### DISCUSSION TITLE
Renode launch

### DISCUSSION THREAD

**Discussion by shannon-Weber** [polarfire-soc/polarfire-soc-documentation#141] [Applications and demos] [answered]
i am unable to launch renode in softconsole.
![image](https://user-images.githubusercontent.com/109544441/179570105-b91b6ab8-4d83-4147-84dd-f00c4b8dd5b8.png)

while following the documentation steps of post installation of soft console in Ubantu 20.04.02 :

> echo "deb https://download.mono-project.com/repo/ubuntu stable-focal main" | sudo tee /etc/apt/sources.list.d/mono-official-stable.list

> sudo apt update

got this warning :

> N: Skipping acquire of configured file 'main/binary-i386/Packages' as repository 'https://download.mono-project.com/repo/ubuntu stable-focal InRelease' doesn't support architecture 'i386'

**Comment by AntonKrug** [comment, marked as answer]
And if you would ignore the warning and will run the `sudo apt install mono-devel mono-complete`, what happens then?

Just in case what is the output when you run `uname -a` ?

**Reply by shannon-Weber** [reply]
After running > sudo apt install mono-devel mono-complete 
Output :
> [WARNING] Couldn't start UI - falling back to telnet mode
> [INFO] Loaded monitor commands from: /home/shannon/Microchip/SoftConsole-v2021.3-7.0.0.599/renode/./scripts/monitor.py
> [INFO] Monitor available in telnet mode on port 1234
> [INFO] Including script: /home/shannon/Microchip/SoftConsole-v2021.3-7.0.0.599/renode-microchip-mods/scripts/miv-generic-board.resc
> [WARNING] Writing to an output GPIO pin #0
> [WARNING] Writing to an output GPIO pin #1
> [WARNING] Writing to an output GPIO pin #2
> Renode has been started successfully and is ready for a gdb connection.
UI is not opening.

> uname -a
Output : 

> Linux shannon-MS-7C96 5.15.0-41-generic #44~20.04.1-Ubuntu SMP Fri Jun 24 13:27:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

**Reply by AntonKrug** [reply]
That 32-bit warning `stable-focal InRelease' doesn't support architecture 'i386'` could be caused because you were adding some 32-bit libs to a 64-bit OS previosuly.

Looks like you have mono installed now, just not sure what version from what 'provider' as distributions have their own too.

What is the output of `mono --version`

Wondering that you should still be able to debug application without having a Renode monitor UI. Just you need to run the correct launcher.
Looks like you are just running the launcher for the renode platform, you want to run the group launchers called `Launch group` which will run platform and then connect with the GDB debug launcher to it.

![image](https://user-images.githubusercontent.com/30338596/179781086-010d06cb-a23a-4146-9837-c3567725fe08.png)


https://mi-v-ecosystem.github.io/SoftConsole-Documentation/SoftConsole-v2021.3/using_softconsole/renode.html

**Reply by shannon-Weber** [reply]
output of mono version -->
> Mono JIT compiler version 6.12.0.182 (tarball Tue Jun 14 22:35:00 UTC 2022)
> Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
> 	TLS:           __thread
> 	SIGSEGV:       altstack
> 	Notifications: epoll
> 	Architecture:  amd64
> 	Disabled:      none
> 	Misc:          softdebug 
> 	Interpreter:   yes
> 	LLVM:          yes(610)
> 	Suspend:       hybrid
> 	GC:            sgen (concurrent by default)

UI is not coming from launch group as well.

**Reply by AntonKrug** [reply]
Wondering if there is something else missing (maybe a package).
You have exactly same version (with the same settings) as I have on my Debian, the provider is the mono project too, so that looks okish.

And when you use launch group, you do not see the Renode GUI, but the SoftConsole starts the beguing sessions? Is the application reaching main? I mean the Renode Window is not showing for some reason but does the rest function correctly?

**Reply by shannon-Weber** [reply]
Yes the application reaching to main, i can debug also.

**Reply by AntonKrug** [reply]
Then I would say your original query "i am unable to launch renode in softconsole" is resolved. And should we close this one?
You have consequent issue as you are not seeing the Renode GUI, could you create a separate question/discution entry for that? So then the discusions here are in form Question->Answer, instead of long threads of various questions grouped under a single entry and a single title.
